### PR TITLE
Add read/write and yaml export to io library

### PR DESCRIPTION
**Pull request by maximejimenez**
Add function required for kubernetes library in order to read/write configuration file and deploy using `kubectl apply`.

_Note:_ I also took the time to add missing typings to already existing function (both params and return value)

### CODE REVIEW

File: nestor_api/lib/io.py [hunk 1]
```diff
@@ -86,12 +86,23 @@ def get_temporary_directory_path(prefix=""):
...
-def remove(file_path):
+def to_yaml(data: dict) -> str:
+    """Converts a dictionary into a valid yaml string"""
+    return yaml.safe_dump(data)
+
+
+def read(file_path: str) -> str:
+    """Read the file content at the given path"""
+    file = open(file_path, "r")
+    return file.read()
```
Comment (benard-g) [inline nestor_api/lib/io.py:102]: You forgot to call `file.close()`.
I recommend to use the `with` structure to ensure this
```python
def read(file_path: str) -> str:
    """Read the file content at the given path"""
    with open(file_path, "r") as file:
        file_content = file.read()
    return file_content
```

File: nestor_api/lib/io.py [hunk 2]
```diff
@@ -34,46 +34,46 @@ def create_temporary_directory(directory_prefix=""):
...
 
 
-def exists(file_path):
+def exists(file_path: str) -> bool:
     """Checks if a file exists"""
     return Path(file_path).exists()
 
 
-def from_yaml(file_path):
+def from_yaml(file_path: str) -> dict:
```
Comment (benard-g) [inline nestor_api/lib/io.py:55]: What do you think of improving this function's name while you're at it ? (I find the `io.from_yaml()` a bit weird personally).
Maybe something similar to "parse_yaml" or "read_yaml" (and then the `to_yaml` you added becomes "write_yaml").

File: nestor_api/lib/io.py [hunk 3]
```diff
@@ -100,3 +111,9 @@ def remove(file_path):
             os.remove(file_path)
         else:
             raise
+
+
+def write(file_path: str, content: str) -> None:
+    """ Write the string content into the file at the given path"""
+    file = open(file_path, "w")
+    file.write(content)
```
Comment (benard-g) [inline nestor_api/lib/io.py:119]: Same comment as above about closing and the `with` structure.

Also, you may need to use `"w+"` to create the file if it does not exists (not a 100% sure on this one).

Comment (maximejimenez) [reply nestor_api/lib/io.py:119]: From what I read, I can see that `+` is only to read as well when opening with w (for writing).

File: nestor_api/lib/io.py [hunk 4]
```diff
@@ -86,12 +83,27 @@ def get_temporary_directory_path(prefix=""):
...
+    """Read the file content at the given path"""
+    with open(file_path, "r") as file:
+        file_content = file.read()
+    return file_content
+
+
+def read_yaml(file_path: str) -> dict:
+    """Returns a dictionary from yaml file path"""
+    with open(file_path, "r") as file_data:
+        yaml_data = file_data.read()
```
Comment (benard-g) [inline nestor_api/lib/io.py:101]: You could call the above `read` function here, do as you prefer.

**Pull request merged**